agent/consul/state/intention.go

```diff
@@ -333,6 +341,32 @@ func (s *Store) IntentionMatch(ws memdb.WatchSet, args *structs.IntentionQueryMa
 			}
 		}
 
+		// If matching an external uri, in addition to the exact match done
+		// above, we also need to check if the uri's domain matches against
+		// any external trust domain intentions. e.g. spiffe://host/path will
+		// match against a trust domain intention with source spiffe://host.
+		if args.Type == structs.IntentionMatchSource && args.SourceType == structs.IntentionSourceExternalURI {
+			uri, err := url.Parse(entry.Name)
+			// If there's an error then nothing will match this url anyway so
+			// we don't search further.
+			if err == nil {
+				params := []interface{}{
```

Comment:
Whats the reason for creating the params list and then expanding it in the `tx.Get` call. I may be missing something but at first glance it looks like this could just be condensed to

```
iter,err := tx.Get(intentionsTableName, string(args.Type), string(structs.IntentionSoureExternalTrustDomain), entry.Namespace, fmt.Sprintf("%s://%s", url.Scheme, uri.Hostname())})
```

That obviously is quite long but maybe the `fmt.Sprintf` could be assigned to a local var.

Essentially its not obvious to me just by looking at this why the `[]interface{}` is useful.

Reply:
No functional reason. It was just to make it more readable. Happy to change it.